File: app/src/main/java/com/example/mrakopediareader/api/parser/PageParser.java
package com.example.mrakopediareader.api.parser;

import com.example.mrakopediareader.api.dto.Page;

import org.json.JSONException;
import org.json.JSONObject;

public class PageParser extends ParserImpl<Page> {
    private static String KEY_TITLE = "title";
    private static String KEY_URL = "url";

    @Override
    public Page fromJsonObject(JSONObject object) throws JSONException {
        final String title = object.getString(KEY_TITLE);
        final String url = object.getString(KEY_URL);
        return new Page(title, url);
    }
}

File: app/src/main/java/com/example/mrakopediareader/api/parser/PageParser.kt
package com.example.mrakopediareader.api.parser

import com.example.mrakopediareader.api.dto.Page
import org.json.JSONObject

internal class PageParser : ParserImpl<Page>() {
    override fun fromJsonObject(jsonObject: JSONObject): Page {
        val title = jsonObject.getString(KEY_TITLE)
        val url = jsonObject.getString(KEY_URL)
        return Page(title, url)
    }

    companion object {
        private const val KEY_TITLE = "title"
        private const val KEY_URL = "url"
    }
}